source/as-stream.ts

```diff
@@ -105,6 +106,15 @@ export default function asStream(options: MergedOptions) {
 		return unpipe(stream);
 	};
 
+	proxy.on('pipe', source => {
+		if (source instanceof IncomingMessage) {
+			options.headers = {
+				...options.headers,
+				...source.headers
```

Comment:
I think the options.headers should take precedence over the default headers from incomingMessage

Reply:
Good spot!